packages/api/portfolioDrafts/funding/createFundingStep.test.ts

```diff
@@ -281,73 +236,59 @@ describe("Individual Clin input validation tests", () => {
       validationMessage: ValidationMessage.TOTAL_GT_OBLIGATED,
     });
   });
-  it("should return error map entries when given Clin has invalid clin number (<4char)", () => {
-    const errors = validateClin(mockClinInvalidClinNumberTooShort);
-    expect(errors).toContainEqual({
-      clinNumber: "1",
-      invalidParameterName: "clin_number",
-      invalidParameterValue: "1",
-      validationMessage: ValidationMessage.INVALID_CLIN_NUMBER,
-    });
-  });
-  it("should return error map entries when given Clin has invalid clin number (>4char)", () => {
-    const errors = validateClin(mockClinInvalidClinNumberTooLong);
-    expect(errors).toContainEqual({
-      clinNumber: "55555",
-      invalidParameterName: "clin_number",
-      invalidParameterValue: "55555",
-      validationMessage: ValidationMessage.INVALID_CLIN_NUMBER,
-    });
-  });
-  it("should return error map entries when given Clin has invalid clin number (all 0s)", () => {
-    const errors = validateClin(mockClinInvalidClinNumberAllZeros);
-    expect(errors).toContainEqual({
-      clinNumber: "0000",
-      invalidParameterName: "clin_number",
-      invalidParameterValue: "0000",
-      validationMessage: ValidationMessage.INVALID_CLIN_NUMBER,
-    });
-  });
   it("should return no error map entries when given Clin has these funding values (obligated=total)", () => {
-    const errors = validateClin(mockClinObligatedEqualsTotal);
+    const errors = validateClin(mockClinObligatedEqualsTotal());
     expect(errors).toStrictEqual([]);
   });
 });
 
-describe("All Clins in Funding Step input validation tests", () => {
+describe("Middy validation checks", function () {
+  it("should return ValidationErrorResponse that includes error_map in response body", async () => {
+    const result = await handler(validRequestBadData, {} as Context, () => null);
+    const responseBody = JSON.parse(result?.body ?? "");
+    expect(result?.statusCode).toEqual(ErrorStatusCode.BAD_REQUEST);
+    expect(responseBody.message).toMatch(/Request failed validation/);
+    expect(responseBody.code).toMatch(/INVALID_INPUT/);
+    // task_order_number is invalid
+    expect(responseBody.error_map[0].instancePath).toEqual("/body/task_orders/0/task_order_number");
+    expect(responseBody.error_map[0].message).toEqual('must match pattern "^[\\d]{13,17}$"');
+    // clin_number is invalid
+    expect(responseBody.error_map[1].instancePath).toEqual("/body/task_orders/0/clins/0/clin_number");
+    expect(responseBody.error_map[1].message).toEqual('must match pattern "(?!^0{4}$)^\\d{4}$"');
+    expect(responseBody.error_map[2].message).toEqual('must match pattern "(?!^0{4}$)^\\d{4}$"');
+    expect(responseBody.error_map[3].message).toEqual('must match pattern "(?!^0{4}$)^\\d{4}$"');
```

Comment:
and also, are we sure we want to guarantee ordering of fields?

Reply:
This test uses middy and `mockFundingStepBadData` which is consistent when it comes to the order the error messages come in. We _could_ remove these from this tests, but we can't do it at the `validateClins` level because these checks were removed (middy covers them).